src/string.cr

```diff
@@ -2389,10 +2389,6 @@ class String
   # this string is less, equal or greater than *other*, optionally in a *case_insensitive*
   # manner.
   #
-  # If *case_insitive* is `false`, this method delegates to `<=>`. Otherwise,
```

Comment:
This should be brought back.

Reply:
Why is it useful to know that this method delegates to `<=>`? Also not just ASCII characters are compared in a case-insensitive way, all characters.

Reply:
Fair 'nuff, good point.